### PR TITLE
pulsar (ticdc): fix some pulsar sink related error 

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: ref #9413 

### What is changed and how it works?

Fix some pulsar sink error that I encountered in testing. 

#### Tests 

 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/ddlsink/mq/ddlproducer/pulsar_ddl_producer.go [hunk 1]
```diff
@@ -53,10 +55,14 @@ func (p *pulsarProducers) SyncBroadcastMessage(ctx context.Context, topic string
...
 
 // SyncSendMessage sends a message
-// partitionNum is not used,pulsar consume all partitions
+// partitionNum is not used, pulsar consume all partitions
 func (p *pulsarProducers) SyncSendMessage(ctx context.Context, topic string,
 	partitionNum int32, message *common.Message,
 ) error {
+	startTime := time.Now()
+	defer func() {
+		log.Info("pulsarProducers SyncSendMessage cost time", zap.Duration("cost", time.Since(startTime)))
```
Comment (CharlesCheung96) [inline cdc/sink/ddlsink/mq/ddlproducer/pulsar_ddl_producer.go:64]: changefeedID should always be output to the log, which greatly reduces the difficulty of debugging

File: cdc/model/changefeed.go [hunk 2]
```diff
@@ -372,6 +372,7 @@ func (info *ChangeFeedInfo) RmUnusedFields() {
 }
 
 func (info *ChangeFeedInfo) rmMQOnlyFields() {
+	log.Info("since the downstream is not a MQ, remove MQ only fields")
```
Comment (CharlesCheung96) [inline cdc/model/changefeed.go:375]: Please output changefeedID.

File: cdc/sink/dmlsink/mq/dmlproducer/pulsar_dml_producer.go [hunk 3]
```diff
@@ -119,7 +120,8 @@ func NewPulsarDMLProducer(
 		failpointCh: failpointCh,
 		errChan:     errCh,
 	}
-
+	log.Info("Pulsar DML producer created", zap.Stringer("changefeed", p.id),
```
Comment (CharlesCheung96) [inline cdc/sink/dmlsink/mq/dmlproducer/pulsar_dml_producer.go:123]:
```suggestion
	log.Info("Pulsar DML producer created", zap.Stringer("changefeed", p.id), zap.String("namespace", changefeedID.Namespace),

```

File: cdc/sink/dmlsink/mq/dmlproducer/pulsar_dml_producer.go [hunk 4]
```diff
@@ -169,7 +171,16 @@ func (p *pulsarDMLProducer) AsyncSendMessage(
 					zap.String("changefeed", p.id.ID),
 					zap.Error(err))
 				mq.IncPublishedDMLFail(topic, p.id.ID, message.GetSchema())
-				p.errChan <- e
+				// use this select to avoid send error to a closed channel
+				// the ctx will always be called before the errChan is closed
```
Comment (CharlesCheung96) [inline cdc/sink/dmlsink/mq/dmlproducer/pulsar_dml_producer.go:175]: Maybe errChan should not be closed?

File: pkg/config/sink.go [hunk 5]
```diff
@@ -488,27 +487,27 @@ func (c *PulsarConfig) validate() (err error) {
 
 // GetBrokerURL get broker url
 func (c *PulsarConfig) GetBrokerURL() string {
```
Comment (CharlesCheung96) [inline pkg/config/sink.go:489]: Get and Set methods should be removed。

File: cdc/sink/dmlsink/mq/dmlproducer/pulsar_dml_producer.go (hunk 3 above)
Comment (asddongmen) [reply cdc/sink/dmlsink/mq/dmlproducer/pulsar_dml_producer.go:123]: The `Stringer` method of `zap` will call the `String` method of `changefeedID` and it already print the namespace of the changefeed.

File: cdc/sink/dmlsink/mq/dmlproducer/pulsar_dml_producer.go (hunk 4 above)
Comment (asddongmen) [reply cdc/sink/dmlsink/mq/dmlproducer/pulsar_dml_producer.go:175]: Then the errChan may be leaked.

File: pkg/config/sink.go [hunk 6]
```diff
@@ -464,12 +464,11 @@ type PulsarConfig struct {
...
-	// This parameter from the sink-uri
-	brokerURL string `toml:"-" json:"-"`
-
-	// parse the sinkURI
-	u *url.URL `toml:"-" json:"-"`
+	// BrokerURL is used to configure service brokerUrl for the Pulsar service.
+	// This parameter is a part of the `sink-uri`. Internal use only.
+	BrokerURL string `toml:"-" json:"-"`
+	// U is the parsed sinkURI. Internal use only.
+	U *url.URL `toml:"-" json:"-"`
```
Comment (CharlesCheung96) [inline pkg/config/sink.go:471]:
```suggestion
	uri *url.URL `toml:"-" json:"-"`
```
The use of URI and URL needs to be uniform.

Comment (sdojjy) [inline pkg/config/sink.go]: U is not a good name.

**Review by sdojjy — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**